tests/unit/net/subresource_integrity.rs

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use net::subresource_integrity::{SriEntry, get_prioritized_hash_function, get_strongest_metadata};
+use net::subresource_integrity::{is_response_integrity_valid, parsed_metadata};
+use net_traits::response::{Response, ResponseBody};
+use servo_url::ServoUrl;
+
+#[test]
+fn test_get_prioritized_hash_function() {
+    let mut algorithm = get_prioritized_hash_function("sha256", "sha256");
+    assert_eq!(algorithm, "".to_owned());
+
+    algorithm = get_prioritized_hash_function("sha256", "sha384");
+    assert_eq!(algorithm, "sha384".to_owned());
+
+    algorithm = get_prioritized_hash_function("sha384", "sha512");
+    assert_eq!(algorithm, "sha512".to_owned());
+}
+
+#[test]
+fn test_parsed_metadata_without_options() {
```

Comment:
Let's add a test for malformed integrity attributes, too. No `-`s, multiple `?`s, etc.